File: stacktots_mobile/lib/services/subscription_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';

class SubscriptionService {
  final String _apiUrl = 'http://localhost:3000/api/subscriptions';

  Future<List<dynamic>> getPlans() async {
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.getString('token');

    final response = await http.get(
      Uri.parse('$_apiUrl/plans'),
      headers: <String, String>{
        'Content-Type': 'application/json; charset=UTF-8',
        'Authorization': 'Bearer $token',
      },
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to load plans');
    }
  }

  Future<void> createSubscription(int planId, String paymentMethodId) async {
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.getString('token');

    final response = await http.post(
      Uri.parse('$_apiUrl/create'),
      headers: <String, String>{
        'Content-Type': 'application/json; charset=UTF-8',
        'Authorization': 'Bearer $token',
      },
      body: jsonEncode(<String, dynamic>{
        'planId': planId,
        'paymentMethodId': paymentMethodId,
      }),
    );

    if (response.statusCode != 200) {
      throw Exception('Failed to create subscription');
    }
  }
}

File: stacktots-frontend/src/app/services/subscription.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SubscriptionService {

  private apiUrl = 'http://localhost:3000/api/subscriptions';

  constructor(private http: HttpClient) { }

  private getAuthHeaders(): HttpHeaders {
    const token = localStorage.getItem('token');
    return new HttpHeaders({
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${token}`
    });
  }

  getPlans(): Observable<any> {
    return this.http.get(`${this.apiUrl}/plans`, { headers: this.getAuthHeaders() });
  }

  createSubscription(planId: number, paymentMethodId: string): Observable<any> {
    return this.http.post(`${this.apiUrl}/create`, { planId, paymentMethodId }, { headers: this.getAuthHeaders() });
  }
}
